libraries/xcode.rb

```diff
@@ -54,3 +54,6 @@ def available_simulator_versions
     end
   end
 end
+
+Chef::Recipe.include(MacOS::XcodeHelpers)
```

Comment:
Why do we need to have the Xcode helpers available in all recipes and resources?

The docs point to using an [`action_class`](https://docs.chef.io/custom_resources.html#action-class) for resource actions. Are we using these in recipes?

Reply:
There's actually a refactor of this using `action_class` coming up in another branch (xcode-idempotence) - I was hoping to clear this up in the meantime. Right now some of these helper methods are used in downstream cookbooks when further tinkering with Xcode installs.

Reply:
In order for helper methods to be used in guards, they must be made available to `Chef::Recipe`. In order for them to be used "inside" a resource (i.e. a property or part of custom resource logic), they must be included in `Chef::Resource`.

You are correct about `action_class`, but that refactor is outside the scope of this PR and upcoming in `feature/xcode-idempotence`.

Reply:
👍 Thank you for reminding me!

Reply:
I personally don't like adding these methods directly to the global namespace of resources & recipes. It clutters the namespace, provides poor encapsulation, and forces the use of longer than needed method names in order to clarify intent. It's much cleaner if the call site can read something like `Xcode.installed?` rather than a global method `xcode_installed?`.